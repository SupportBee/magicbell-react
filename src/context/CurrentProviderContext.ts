import { createContext } from 'react';

export const NO_PROVIDER = 'MAGICBELL_NO_PROVIDER';

const CurrentProviderContext = createContext<String>(NO_PROVIDER);
export default CurrentProviderContext;
