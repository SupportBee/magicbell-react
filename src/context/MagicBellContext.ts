import { createContext, useContext } from 'react';

export interface IMagicBellContext {
  images?: Partial<{
    emptyInboxUrl: string;
  }>;
}

const MagicBellContext = createContext<IMagicBellContext>({} as IMagicBellContext);
export default MagicBellContext;

const useMagicBellContext = () => useContext(MagicBellContext);
export { useMagicBellContext };
