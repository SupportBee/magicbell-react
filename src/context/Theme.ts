export interface IElementTheme {
  backgroundColor: string;
  backgroundOpacity: number;
  borderRadius: string;
  fontFamily: string;
  fontSize: string;
  textAlign: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
  textColor: string;
  textTransform: 'none' | 'capitalize' | 'uppercase' | 'lowercase' | 'initial' | 'inherit';
}

export interface IMagicBellTheme {
  icon: {
    borderColor: string;
    width: string;
  };
  unseenBadge: IElementTheme;
  header: IElementTheme;
  container: IElementTheme;
  footer: IElementTheme;
  notification: {
    default: IElementTheme;
    unread: IElementTheme;
    unseen: IElementTheme;
  };
}

const blue = '#3498F4';
const red = '#DF4759';
const black = '#3A424D';

// prettier-ignore
const fontFamily = '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif';
const fontSize = '14px';
const textAlign = 'left';
const textTransform = 'none';

export const defaultTheme = {
  icon: {
    borderColor: blue,
    width: '24px',
  },
  header: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  footer: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  unseenBadge: {
    backgroundColor: red,
    backgroundOpacity: 1,
    borderRadius: '2px',
    fontFamily,
    fontSize: '10px',
    textAlign: 'center',
    textColor: 'white',
    textTransform,
  },
  container: {
    backgroundColor: '#FFFFFF',
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: '14px',
    textAlign,
    textColor: black,
    textTransform,
  },
  notification: {
    default: {
      backgroundColor: blue,
      backgroundOpacity: 0.1,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unread: {
      backgroundColor: blue,
      backgroundOpacity: 0,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unseen: {
      backgroundColor: blue,
      backgroundOpacity: 0.05,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
  },
} as IMagicBellTheme;
