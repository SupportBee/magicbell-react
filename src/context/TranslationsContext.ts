import { createContext, useContext } from 'react';

const TranslationsContext = createContext<any>({});
export default TranslationsContext;

const TranslationsProvider = TranslationsContext.Provider;
const useTranslations = () => useContext(TranslationsContext);
export { TranslationsProvider, useTranslations };
