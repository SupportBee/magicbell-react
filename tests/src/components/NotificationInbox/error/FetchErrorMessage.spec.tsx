import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import FetchErrorMessage from '../../../../../src/components/NotificationInbox/error/FetchErrorMessage';
import { MagicBellThemeProvider } from '../../../../../src/context/MagicBellThemeContext';
import { defaultTheme } from '../../../../../src/context/Theme';

describe('components', () => {
  describe('FetchErrorMessage', () => {
    let view: RenderResult;

    beforeEach(() => {
      Object.defineProperty(window.navigator, 'onLine', { value: true, writable: true });
    });

    afterEach(() => {
      view.unmount();
    });

    describe('render', () => {
      describe('the navigator has an internet connection', () => {
        it('renders an API error message', () => {
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-expect-error
          window.navigator.onLine = true;
          view = render(
            <MagicBellThemeProvider value={defaultTheme}>
              <FetchErrorMessage />
            </MagicBellThemeProvider>,
          );

          expect(view.container).toMatchSnapshot();
        });
      });

      describe('the navigator does not have an internet connection', () => {
        it('renders a connection error message', () => {
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-expect-error
          window.navigator.onLine = false;
          view = render(
            <MagicBellThemeProvider value={defaultTheme}>
              <FetchErrorMessage />
            </MagicBellThemeProvider>,
          );

          expect(view.container).toMatchSnapshot();
        });
      });
    });
  });
});
