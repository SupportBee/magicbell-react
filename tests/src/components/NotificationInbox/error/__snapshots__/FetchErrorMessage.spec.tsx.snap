// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components FetchErrorMessage render the navigator does not have an internet connection renders a connection error message 1`] = `
.emotion-0 {
  padding: 3em!important;
  text-align: center;
  color: #3A424D;
}

.emotion-0 svg {
  margin: 2em auto;
  height: 200px;
}

<div>
  <div
    class="emotion-0"
  >
    <div>
      <p>
        <span>
          Hmm, we’re unable to connect to the internet.
          <br />
          Please check your connection.
        </span>
      </p>
      <svg
        fill="currentColor"
        height="93"
        viewBox="0 0 120 93"
        width="120"
      >
        <path
          d="M13.5135 93H106.486C113.919 93 120 86.9871 120 79.6379V27.5259V13.3621C120 6.01293 113.919 0 106.486 0H13.5135C6.08108 0 0 6.01293 0 13.3621V27.3922V79.5043C0 86.9871 6.08108 93 13.5135 93ZM2.7027 13.3621C2.7027 7.48276 7.56757 2.67241 13.5135 2.67241H106.486C112.432 2.67241 117.297 7.48276 117.297 13.3621V24.7198H2.7027V13.3621ZM2.7027 27.3922H117.297V79.5043C117.297 85.3836 112.432 90.194 106.486 90.194H13.5135C7.56757 90.194 2.7027 85.3836 2.7027 79.5043V27.3922Z"
        />
        <path
          d="M13.6485 5.87933C9.05392 5.87933 5.40527 9.48709 5.40527 14.0302C5.40527 18.5733 9.05392 22.1811 13.6485 22.1811C18.2431 22.1811 21.8918 18.5733 21.8918 14.0302C21.8918 9.48709 18.108 5.87933 13.6485 5.87933ZM13.6485 19.375C10.5404 19.375 8.10798 16.9698 8.10798 13.8966C8.10798 10.8233 10.5404 8.41813 13.6485 8.41813C16.7566 8.41813 19.1891 10.8233 19.1891 13.8966C19.1891 16.9698 16.6215 19.375 13.6485 19.375Z"
        />
        <path
          d="M32.4322 5.87933C27.8376 5.87933 24.189 9.48709 24.189 14.0302C24.189 18.5733 27.8376 22.1811 32.4322 22.1811C37.0268 22.1811 40.6755 18.5733 40.6755 14.0302C40.6755 9.48709 36.8917 5.87933 32.4322 5.87933ZM32.4322 19.375C29.3241 19.375 26.8917 16.9698 26.8917 13.8966C26.8917 10.8233 29.3241 8.41813 32.4322 8.41813C35.5403 8.41813 37.9727 10.8233 37.9727 13.8966C37.9727 16.9698 35.4052 19.375 32.4322 19.375Z"
        />
        <path
          d="M51.2164 5.87933C46.6218 5.87933 42.9731 9.48709 42.9731 14.0302C42.9731 18.5733 46.6218 22.1811 51.2164 22.1811C55.811 22.1811 59.4596 18.5733 59.4596 14.0302C59.4596 9.48709 55.6758 5.87933 51.2164 5.87933ZM51.2164 19.375C48.1083 19.375 45.6758 16.9698 45.6758 13.8966C45.6758 10.8233 48.1083 8.41813 51.2164 8.41813C54.3245 8.41813 56.7569 10.8233 56.7569 13.8966C56.7569 16.9698 54.1894 19.375 51.2164 19.375Z"
        />
        <path
          d="M42.4924 57.2198C43.2979 62.379 47.594 66.3 52.8299 66.3H56.5219C57.4617 66.3 58.2001 65.5433 58.2001 64.5803V46.4198C58.2001 45.4567 57.4617 44.7 56.5219 44.7H52.8971C47.6611 44.7 43.2979 48.621 42.5595 53.7803H38.2783C37.3385 53.7803 36.6001 54.537 36.6001 55.5C36.6001 56.4631 37.3385 57.2198 38.2783 57.2198H42.4924ZM52.9642 48.1395H54.978V62.9293H52.9642C49.0037 62.8605 45.7816 59.5586 45.7816 55.5C45.6473 51.4414 49.0037 48.1395 52.9642 48.1395Z"
        />
        <path
          d="M60.8833 58.3204C60.0527 58.3204 59.4001 59.0771 59.4001 60.0401C59.4001 61.0032 60.0527 61.7599 60.8833 61.7599H64.3242V64.5803C64.3242 65.5433 64.9768 66.3 65.8073 66.3H69.0109C73.6384 66.3 77.4945 62.379 78.1471 57.2198H81.917C82.7476 57.2198 83.4001 56.4631 83.4001 55.5C83.4001 54.537 82.7476 53.7803 81.917 53.7803H78.2064C77.4945 48.621 73.6977 44.7 69.0703 44.7H65.8073C64.9768 44.7 64.3242 45.4567 64.3242 46.4198V49.2401H60.8833C60.0527 49.2401 59.4001 49.9968 59.4001 50.9599C59.4001 51.9229 60.0527 52.6796 60.8833 52.6796H64.3242V58.3204H60.8833ZM67.1718 48.0707H68.9516C72.4518 48.0707 75.2995 51.3726 75.2995 55.4312C75.2995 59.4898 72.4518 62.7917 68.9516 62.7917H67.1718V48.0707Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`components FetchErrorMessage render the navigator has an internet connection renders an API error message 1`] = `
.emotion-0 {
  padding: 3em!important;
  text-align: center;
  color: #3A424D;
}

.emotion-0 svg {
  margin: 2em auto;
  height: 200px;
}

<div>
  <div
    class="emotion-0"
  >
    <div>
      <p>
        <span>
          We can’t seem to retrieve your notifications.
          <br />
          Please check back soon.
        </span>
      </p>
      <svg
        fill="currentColor"
        height="94"
        viewBox="0 0 120 94"
        width="120"
      >
        <path
          d="M13.5135 94H106.486C113.919 94 120 87.9224 120 80.4942V27.8218V13.5057C120 6.07759 113.919 0 106.486 0H13.5135C6.08108 0 0 6.07759 0 13.5057V27.6868V80.3592C0 87.9224 6.08108 94 13.5135 94ZM2.7027 13.5057C2.7027 7.56322 7.56757 2.70115 13.5135 2.70115H106.486C112.432 2.70115 117.297 7.56322 117.297 13.5057V24.9856H2.7027V13.5057ZM2.7027 27.6868H117.297V80.3592C117.297 86.3017 112.432 91.1638 106.486 91.1638H13.5135C7.56757 91.1638 2.7027 86.3017 2.7027 80.3592V27.6868Z"
        />
        <path
          d="M13.6486 5.9425C9.05404 5.9425 5.4054 9.58906 5.4054 14.181C5.4054 18.773 9.05404 22.4195 13.6486 22.4195C18.2432 22.4195 21.8919 18.773 21.8919 14.181C21.8919 9.58906 18.1081 5.9425 13.6486 5.9425ZM13.6486 19.5833C10.5405 19.5833 8.1081 17.1523 8.1081 14.046C8.1081 10.9396 10.5405 8.5086 13.6486 8.5086C16.7567 8.5086 19.1892 10.9396 19.1892 14.046C19.1892 17.1523 16.6216 19.5833 13.6486 19.5833Z"
        />
        <path
          d="M32.4325 5.9425C27.8379 5.9425 24.1892 9.58906 24.1892 14.181C24.1892 18.773 27.8379 22.4195 32.4325 22.4195C37.027 22.4195 40.6757 18.773 40.6757 14.181C40.6757 9.58906 36.8919 5.9425 32.4325 5.9425ZM32.4325 19.5833C29.3243 19.5833 26.8919 17.1523 26.8919 14.046C26.8919 10.9396 29.3243 8.5086 32.4325 8.5086C35.5406 8.5086 37.973 10.9396 37.973 14.046C37.973 17.1523 35.4054 19.5833 32.4325 19.5833Z"
        />
        <path
          d="M51.2163 5.9425C46.6217 5.9425 42.973 9.58906 42.973 14.181C42.973 18.773 46.6217 22.4195 51.2163 22.4195C55.8109 22.4195 59.4595 18.773 59.4595 14.181C59.3244 9.58906 55.6757 5.9425 51.2163 5.9425ZM51.2163 19.5833C48.1082 19.5833 45.6757 17.1523 45.6757 14.046C45.6757 10.9396 48.1082 8.5086 51.2163 8.5086C54.3244 8.5086 56.7568 10.9396 56.7568 14.046C56.6217 17.1523 54.1892 19.5833 51.2163 19.5833Z"
        />
        <path
          d="M18.9188 75.6321C19.1891 75.9022 19.5945 76.0373 19.8648 76.0373C20.135 76.0373 20.5405 75.9022 20.8107 75.6321L36.4864 59.9655C37.0269 59.4252 37.0269 58.6149 36.4864 58.0746L20.8107 42.408C20.2702 41.8678 19.4594 41.8678 18.9188 42.408C18.3783 42.9482 18.3783 43.7586 18.9188 44.2988L33.6486 59.0201L18.9188 73.7413C18.3783 74.2815 18.3783 75.0919 18.9188 75.6321Z"
        />
        <path
          d="M47.5675 60.3707H72.2973C73.1081 60.3707 73.6486 59.8305 73.6486 59.0201C73.6486 58.2098 73.1081 57.6696 72.2973 57.6696H47.5675C46.7567 57.6696 46.2162 58.2098 46.2162 59.0201C46.2162 59.8305 46.8919 60.3707 47.5675 60.3707Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
